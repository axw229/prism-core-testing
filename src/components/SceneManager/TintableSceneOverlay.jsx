// @flow
import React, { PureComponent } from 'react'
import CircleLoader from '../Loaders/CircleLoader/CircleLoader'

type Props = {
  type: 'LOADING' | 'ERROR' | 'MESSAGE',
  message?: string,
  color: string
}

class TintableSceneOverlay extends PureComponent<Props> {
  static TYPES = {
    LOADING: 'LOADING',
    ERROR: 'ERROR',
    MESSAGE: 'MESSAGE'
  }

  static CLASS_NAMES = {
    [TintableSceneOverlay.TYPES.MESSAGE]: {
      BASE: 'prism-scene-manager__scene__overlay prism-scene-manager__scene__overlay--message',
      CONTENT: 'prism-scene-manager__scene__overlay__content'
    },
    [TintableSceneOverlay.TYPES.LOADING]: {
      BASE: 'prism-scene-manager__scene__overlay prism-scene-manager__scene__overlay--loading',
      CONTENT: 'prism-scene-manager__scene__overlay__content'
    },
    [TintableSceneOverlay.TYPES.ERROR]: {
      BASE: 'prism-scene-manager__scene__overlay prism-scene-manager__scene__overlay--error',
      CONTENT: 'prism-scene-manager__scene__overlay__content'
    }
  }

  render () {
    const { type, message } = this.props
    let { color } = this.props

    return (
      <div className={TintableSceneOverlay.CLASS_NAMES[type].BASE}>
        <span className={TintableSceneOverlay.CLASS_NAMES[type].CONTENT}>
          {(type === TintableSceneOverlay.TYPES.LOADING) && (
            <CircleLoader color={color} />
          )}
          {!!message && message}
        </span>
      </div>
    )
  }
}

export default TintableSceneOverlay
