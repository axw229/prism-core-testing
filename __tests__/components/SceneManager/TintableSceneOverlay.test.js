import React from 'react'
import { shallow } from 'enzyme'
import TintableSceneOverlay from 'src/components/SceneManager/TintableSceneOverlay'

const defaultProps = {
  type: 'LOADING' | 'ERROR' | 'MESSAGE'
}

const createOverlay = (props = {}) => {
  return shallow(<TintableSceneOverlay {...defaultProps} {...props} />)
}

// snapshot testing
describe('Snapshot Testing', () => {
  it('component should match snapshot', () => {
    const wrapper = createOverlay({ type: 'MESSAGE' })
    expect(wrapper).toMatchSnapshot()
  })
})

// rendering test
describe('testing Tintable Scene Overlay', () => {
  it('message should rendering correctly', () => {
    const wrapper = createOverlay({ type: 'MESSAGE', message: 'testing' })
    expect(wrapper.find('span').exists()).toBe(true)
    expect(wrapper.find('span').text()).toEqual('testing')
  })

  it('loader should rendering correctly', () => {
    const wrapper = createOverlay({ type: 'LOADING', loaderColor: 'blue' })
    expect(wrapper.find('CircleLoader').exists()).toBe(true)
  })
})
