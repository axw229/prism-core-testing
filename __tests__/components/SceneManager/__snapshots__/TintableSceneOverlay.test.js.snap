// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing component should match snapshot 1`] = `
<div
  className="prism-scene-manager__scene__overlay prism-scene-manager__scene__overlay--message"
>
  <span
    className="prism-scene-manager__scene__overlay__content"
  />
</div>
`;
